### PR TITLE
Multiple serializers

### LINKED ISSUE #65: Multiple serializers for one model
Hi all!
As I already suggested in #59, we can use `tag` option for splitting serializers. I propose this API:
```ruby
class HumanSerializer < Surrealist::Serializer
  json_schema do
    { name: String, name_length: Integer }
  end

  def name_length
    name.length
  end
end

class SmallHumanSerializer < Surrealist::Serializer
  json_schema do
    { name: String }
  end
end

class Human
  include Surrealist

  attr_reader :name

  surrealize_with HumanSerializer # default
  surrealize_with SmallHumanSerializer, tag: :small

  def initialize(name)
    @name = name
  end
end
```

```ruby
# default behaviour
Human.new('John').surrealize(camelize: true)
# => '{"name": "John", "nameLength": 4}'

# using tags
Human.new('John').surrealize(camelize: true, tag: :small)
# => '{"name": "John"}'
```

I can realize this functionality and I know that it will be very useful feature.

### PR DESCRIPTION
**Pull request by nulldef**
Resolve #65

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 [![Gem Version](https://badge.fury.io/rb/surrealist.svg)](https://rubygems.org/gems/surrealist)
 
 ![Surrealist](surrealist-icon.png)
- 
+
```
Comment (nesaulov) [inline README.md]: Could you please rollback changes concerning whitespaces here? They are necessary for proper code displaying.

File: lib/surrealist.rb [hunk 2]
```diff
@@ -148,7 +148,8 @@ def build_schema(instance:, **args)
     #   provided in the object's class. Values will be taken from the return values
     #   of appropriate methods from the object.
     def __build_schema(object, **args)
-      if (serializer = Surrealist::VarsHelper.find_serializer(object.class))
+      serializer = Surrealist::VarsHelper.find_serializer(object.class, tag: args[:tag])
+      if serializer
```
Comment (nesaulov) [inline lib/surrealist.rb]: I think you forgot to apply new feature to `#build_schema` method.

File: lib/surrealist/class_methods.rb [hunk 3]
```diff
@@ -98,9 +98,9 @@ def delegate_surrealization_to(klass)
     # @param [Class] klass a class that should inherit form Surrealist::Serializer
     #
     # @raise ArgumentError if Surrealist::Serializer is not found in the ancestors chain
-    def surrealize_with(klass)
+    def surrealize_with(klass, tag: :default)
```
Comment (nesaulov) [inline lib/surrealist/class_methods.rb]: Maybe we should use `DEFAULT_TAG` here also?

File: lib/surrealist/vars_helper.rb [hunk 4]
```diff
@@ -41,18 +43,24 @@ def set_schema(klass, hash)
...
       #
       # @param [Class] klass a class to check
+      # @param [Symbol] tag a tag associated with serializer
       #
       # @return [Class | nil]
-      def find_serializer(klass)
-        klass.instance_variable_get(SERIALIZER_CLASS)
+      def find_serializer(klass, tag: nil)
+        tag ||= DEFAULT_TAG
+        klass.instance_variable_get(SERIALIZER_CLASSES).try(:[], tag.to_sym)
```
Comment (nesaulov) [inline lib/surrealist/vars_helper.rb]: We don't have ActiveSupport in here 🙃

Comment (nesaulov) [reply lib/surrealist/vars_helper.rb]: You can use `return` + `fetch(tag.to_sym, nil)` here I guess (safe navigation would be better, but it's not available in ruby 2.2)

File: lib/surrealist.rb (hunk 2 above)
Comment (nulldef) [reply lib/surrealist.rb]: There is no external serializer support for this method

Comment (nesaulov) [reply lib/surrealist.rb]: Yeah, I guess I missed it in #61.

File: lib/surrealist/vars_helper.rb [hunk 5]
```diff
@@ -41,18 +43,26 @@ def set_schema(klass, hash)
...
       # @param [Class] klass a class to check
+      # @param [Symbol] tag a tag associated with serializer
       #
       # @return [Class | nil]
-      def find_serializer(klass)
-        klass.instance_variable_get(SERIALIZER_CLASS)
+      def find_serializer(klass, tag: nil)
+        tag ||= DEFAULT_TAG
+        hash = klass.instance_variable_get(SERIALIZER_CLASSES)
+        return if hash.nil?
```
Comment (nesaulov) [inline lib/surrealist/vars_helper.rb]: Maybe `hash && hash.fetch()` ?

File: spec/multiple_serializers_spec.rb [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      let(:expectation) do
+        [
+          { id: 1, title: 'Ruby is dead' },
+          { id: 1, title: 'Ruby is dead' },
+          { id: 1, title: 'Ruby is dead' },
+        ]
+      end
+
+      it do
+        json = Surrealist.surrealize_collection(collection, tag: :short)
```
Comment (nesaulov) [inline spec/multiple_serializers_spec.rb]: Maybe extract this to rspec's `let`, so we can make it a one-liner?
Or use `specify` instead of `it`

**Comment by nesaulov** [comment]
@nulldef one more thing: we need an error message for the case when the wrong tag is passed. From your spec right now:
```ruby
Surrealist.surrealize_collection([post1, post2], tag: :kek)
  Surrealist::UnknownSchemaError: Can't serialize Post - no schema was provided.
from /Users/nesaulov/Code/ruby/opensource/surrealist/lib/surrealist/exception_raiser.rb:49:in `raise_unknown_schema!'
```
The error message is wrong and misleading. I suggest something like "The tag specified (#{tag}) has no corresponding serializer".

**Comment by nesaulov** [comment]
Thank you, @nulldef! 🎉

**Pull request merged**